changelog/10991.improvement.rst

```diff
@@ -0,0 +1,2 @@
+Added handling of ``%f`` directive to print microseconds in log format options, such as ``log-date-format``.
+Added helper :class:`pytest.logging.DatetimeFormatter <_pytest.logging.DatetimeFormatter>` which overrides :class:`logging.Formatter` behaviour to print log records with a microsecond-aware function :func:`datetime.strftime` instead of :func:`time.strftime`.
```

Comment:
While I appreciate the added detail, I suggest we remove this line because it is an internal implementation detail, and we do not plan to make `DatetimeFormatter` public. :+1: 

```suggestion
```